### PR TITLE
Layers: Remove VSync from CustomElement

### PR DESCRIPTION
**Pull request by jridgewell**
PR 1 for Remove direct VSync access.

Part of #12748.

### CODE REVIEW

File: src/custom-element.js [hunk 1]
```diff
@@ -550,13 +550,13 @@ function createBaseCustomElementClass(win) {
...
-            if (this.isLoadingEnabled_()) {
+          if (this.resources_) {
+            this.getResources().mutateElement(this, () => {
               this.prepareLoading_();
-            }
-          });
+            });
+          } else {
+            this.prepareLoading_();
+          }
```
Comment (choumx) [inline src/custom-element.js]: A helper function would be clearer. You can also use it on line 1547 below.

```js
this.mutateIfConnected_(() => this.prepareLoading_());

// ...

mutateIfConnected(mutator) {
  if (this.resources_) {
    this.getResources().mutateElement(this, mutator);
  } else {
    mutator();
  }
}
```

File: test/functional/test-custom-element.js [hunk 2]
```diff
@@ -1923,16 +1915,15 @@ describes.realWin('CustomElement Overflow Element', {amp: true}, env => {
     expect(overflowElement.onclick).to.not.exist;
   });
 
-  it('should force change size when clicked', () => {
+  it.only('should force change size when clicked', () => {
```
Comment (choumx) [inline test/functional/test-custom-element.js]: Remove `only`.

File: test/functional/test-custom-element.js [hunk 3]
```diff
@@ -1805,28 +1794,28 @@ describes.realWin('CustomElement', {amp: true}, env => {
...
-        vsyncTasks.shift()();
-        expect(element.loadingContainer_).to.have.class('amp-hidden');
-        expect(element.loadingElement_).to.not.have.class('amp-active');
-
-        // Second vsync should perform cleanup.
-        vsyncTasks.shift()();
-        expect(element.loadingContainer_).to.be.null;
-      });
-    });
+    // TODO(jridgewell): fix this test
```
Comment (choumx) [inline test/functional/test-custom-element.js]: For now or the future?

File: src/custom-element.js (hunk 1 above)
Comment (jridgewell) [reply src/custom-element.js]: Done.

File: test/functional/test-custom-element.js (hunk 3 above)
Comment (jridgewell) [reply test/functional/test-custom-element.js]: Sometime in the future.

File: test/functional/test-custom-element.js (hunk 2 above)
Comment (jridgewell) [reply test/functional/test-custom-element.js]: Done.

File: src/custom-element.js [hunk 4]
```diff
@@ -1650,6 +1645,20 @@ function createBaseCustomElementClass(win) {
...
       }
     }
+
+    /**
+     * Mutates the element using resources if available.
+     *
+     * @param {function()} mutator
+     * @param {?Element=} opt_element
+     */
+    mutateWithConnected_(mutator, opt_element) {
```
Comment (choumx) [inline src/custom-element.js]: I like `mutateOrInvoke_` or `scheduleMutateIfNecessary_`  better.

**Comment by cathyxz** [comment]
This test flaked: 
```
DESCRIBE => Rendering of amp-img
  IT => should respect media queries
    ✗ expected 'i-amphtml-element i-amphtml-layout-responsive i-amphtml-layout-size-defined i-amphtml-layout' to match /i-amphtml-hidden-by-media-query/
```
Can we just aggressively skip flaking tests in the PRs where they flake? The build is broken so often right now. =(

**Comment by cathyxz** [comment]
Oh nvm I guess you fixed it.

**Pull request merged**